### PR TITLE
Spout: New recipe

### PR DESCRIPTION
**Pull request by win8linux**
*Spout* is a simple game originally developed by kuni wherein the player flies through an endless cave in a box-shaped ship and must use the ship's exhaust to clear the way.

This recipe is based on a [fork](https://github.com/win8linux/spout/tree/patched) of the [Unix version by Nick White](https://njw.me.uk/spout/) that integrates the following patches from Debian:

* [Fix hardening errors from unused result](https://salsa.debian.org/smlx-guest/spout/-/blob/debian/sid/debian/patches/0002-Fix-hardening-errors-from-unused-result.patch)
* [Fix call to open with missing argument](https://salsa.debian.org/smlx-guest/spout/-/blob/debian/sid/debian/patches/0003-Fix-call-to-open-with-missing-argument.patch)
* [Fix segfault on 64-bit platforms due to incorrect types](https://salsa.debian.org/smlx-guest/spout/-/blob/debian/sid/debian/patches/0004-Fix-segfault-on-64-bit-platforms-due-to-incorrect-ty.patch)
* [Remove hard-coded library search path /usr/lib](https://salsa.debian.org/smlx-guest/spout/-/blob/debian/sid/debian/patches/0007-Remove-hard-coded-library-search-path-usr-lib.patch)
* [Remove redundant out-of-range comparison](https://salsa.debian.org/smlx-guest/spout/-/blob/debian/sid/debian/patches/0008-Remove-redundant-out-of-range-comparison.patch)
* [Add exit instructions on scroll screen](https://salsa.debian.org/smlx-guest/spout/-/blob/debian/sid/debian/patches/0009-Add-exit-instructions-on-scroll-screen.patch)

Currently a work in progress, as the recipe fails to build for now.

### CODE REVIEW

**Comment by korli** [comment]
> > Only happens on Haiku AFAIK, since it does build successfully on Linux 64-bit.
> 
> Unfortunate, time.h is included for a few types. Please submit a bug report.

As a workaround you could patch spout.c to use another local variable name as time.

File: games-arcade/spout/spout-1.4b.recipe [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+Esc: Pause
+Shift-Esc: Quit"
+HOMEPAGE="https://njw.me.uk/spout/"
+COPYRIGHT="2001-2006 Kuni, 2010 Nick White"
+LICENSE="MIT"
+REVISION="1"
+SOURCE_URI="https://github.com/win8linux/spout/archive/$portVersion.tar.gz"
+CHECKSUM_SHA256="3f3c1d9e9c67e5512e786fc94eaeaf3c6adabd5abe08653adbaff4bb03cfaf66"
+
+ARCHITECTURES="all"
```
Comment (Begasus) [inline games-arcade/spout/spout-1.4b.recipe:26]: Doesn't build for gcc2, so you need to enable secondary architecture for 32bit

**Comment by Begasus** [comment]
> I've tried replacing `time` with another variable name, but the same error occurs still.

Just replace `time` with `_time` (in all cases in spout.c) and the build will work ;)

**Comment by win8linux** [comment]
Thanks, can confirm that doing so does indeed let the build finish. However, the game itself is unplayable with graphical aberrations and instantly going to the game over screen upon user interaction. Will prolly need to investigate further in the future, see what needs to be fixed.

**Comment by Begasus** [comment]
> Should I still upload those changes remotely?

Yes please, the build in itself should be OK with those changes, we'll hold off on merging until you can confirm this is running as expected.

**Comment by Begasus** [comment]
@win8linux any progress on this? If not maybe merge as disabled for all arch's?

**Comment by win8linux** [comment]
I'll just close this, since apparently even Debian has stopped shipping this in the repos of the last couple stable versions due to how broken it has become.

**Comment by Begasus** [comment]
Thanks for getting back on this, if need be someone could proceed from this to check it out. +1

**Pull request closed without merging**